controllers/front/callbacknotification.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * AltaPay module for PrestaShop
+ *
+ * Copyright © 2020 AltaPay. All rights reserved.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+require_once _PS_MODULE_DIR_ . '/altapay/lib/AltapayCallbackHandler.class.php';
+require_once _PS_MODULE_DIR_ . '/altapay/helpers.php';
+
+class AltapayCallbacknotificationModuleFrontController extends ModuleFrontController
+{
+    /**
+     * Method to follow when callback notification is being triggered
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function postProcess()
+    {
+        try {
+            $xml             = Tools::getValue('xml');
+            $callbackHandler = new AltapayCallbackHandler();
+            $response        = $callbackHandler->parseXmlResponse($xml);
+
+            $shopOrderId = $response->getPrimaryPayment()->getShopOrderId();
+
+            $fp = fopen(_PS_MODULE_DIR_ . '/altapay/controllers/front/lock.txt', 'r');
+            flock($fp, LOCK_EX);
+
+            // Load the cart
+            $cart = getCartFromUniqueId($shopOrderId);
+            if (!Validate::isLoadedObject($cart)) {
+                $this->unlock($fp);
+                die('Could not load cart - exiting');
```

Comment:
About 10 of thease left
```suggestion
                exit('Could not load cart - exiting');
```